new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCoherenceTests.java

```diff
@@ -0,0 +1,403 @@
+    logger.info("Delete domain custom resource in namespace {0}", domainNamespace);
+    assertDoesNotThrow(() -> deleteDomainCustomResource(domainUid, domainNamespace),
+        "deleteDomainCustomResource failed with ApiException");
+    logger.info("Deleted Domain Custom Resource " + domainUid + " from " + domainNamespace);
+  }
+
+  /**
+   * Create a WebLogic domain with a Coherence cluster and deploying it using WDT
+   * Test rolling restart of Coherence managed servers and verify
+   * that data are not lost during a domain restart.
+   */
+  @Test
+  @DisplayName("Create domain with a Coherence cluster using WDT and test rolling restart")
+  @Slow
+  @MustNotRunInParallel
+  public void testRollingRestart() {
+    final String successMarker = "CACHE-SUCCESS";
+
+    // create and verify WebLogic domain image using model in image with model files
+    String miiImage = createAndVerifyDomainImage();
+
+    // create and verify a WebLogic domain with a Coherence cluster
+    createAndVerifyDomain(miiImage);
+
+    // build the Coherence proxy client program in the server pods
+    // which load and verify the cache
+    copyCohProxyClientAppToPods();
+
+    // run the Coherence proxy client to load the cache
+    String serverName = managedServerPrefix + "1";
+    final ExecResult execResult1 = assertDoesNotThrow(
+        () -> runCoherenceProxyClient(serverName, OP_CACHE_LOAD),
+        String.format("Failed to call Coherence proxy client in pod %s, namespace %s",
+            serverName, domainNamespace));
+
+    assertAll("Check that the cache loaded successfully",
+        () -> assertTrue(execResult1.exitValue() == 0, "Failed to load the catche"),
+        () -> assertTrue(execResult1.stdout().contains(successMarker), "Failed to load the catche")
+    );
+
+    logger.info("\n Coherence proxy client {0} returns {1} \n ",
+        OP_CACHE_LOAD, execResult1.stdout());
+
+    // patch domain to rolling restart it by change restartVersion
+    roolingRestartDomainAndVerify();
+
+    // build the Coherence proxy client program in the server pods
+    // which load and verify the cache
+    copyCohProxyClientAppToPods();
+
+    // run the Coherence proxy client to verify the cache contents
+    final ExecResult execResult2 = assertDoesNotThrow(
+        () -> runCoherenceProxyClient(serverName, OP_CACHE_VALIDATE),
+        String.format("Failed to call Coherence proxy client in pod %s, namespace %s",
+            serverName, domainNamespace));
+
+    assertAll("Check that the cache loaded successfully",
+        () -> assertTrue(execResult1.exitValue() == 0, "Failed to validate the cache"),
+        () -> assertTrue(execResult2.stdout().contains(successMarker), "Failed to validate the cache")
+    );
+
+    logger.info("\n Coherence proxy client {0} returns {1} \n ",
+        OP_CACHE_VALIDATE, execResult2.stdout());
+
+    logger.info("SUCCESS --- testRollingRestart");
+  }
+
+  private void copyCohProxyClientAppToPods() {
```

Comment:
Check if you can use buildApplication in BuildApplication.java to build your app?

Reply:
I actually looked at this method when coding. My understanding is buildApplication is just zip the app. What we need here is 1) create a .gar file, 2) create WDT archive, seems not meet the requirement.

Reply:
Not true. buildApplication copies your app dir (zips it) into a pod with weblogic install, builds the app and then copies it out of the pod. Please check again. Both Sankar and I use this to build apps.

Reply:
my mistake. I am going to try to use buildApplication in BuildApplication.java

Reply:
1. currently, what the test is doing is that app and script are copied to, build and run in MS pods directly
2. using buildApplication method, the test needs to copy app and script to a temp pod and build the app there, and then m=need to copy app and script to MS pods and run it there in MS pods

To keep it simple, decided to do what the test is doing now. So no changes here

Reply:
I still think we should reuse existing code, but since there seem to be some issues right now using it, I guess it is fine.